presto-main/src/main/java/com/facebook/presto/operator/repartition/ArrayBlockEncodingBuffer.java

```diff
@@ -201,17 +204,19 @@ public String toString()
     }
 
     @Override
-    protected void setupDecodedBlockAndMapPositions(DecodedBlockNode decodedBlockNode)
+    protected void setupDecodedBlockAndMapPositions(DecodedBlockNode decodedBlockNode, int partitionBufferCapacity)
     {
         requireNonNull(decodedBlockNode, "decodedBlockNode is null");
 
         decodedBlockNode = mapPositionsToNestedBlock(decodedBlockNode);
         ColumnarArray columnarArray = (ColumnarArray) decodedBlockNode.getDecodedBlock();
         decodedBlock = columnarArray.getNullCheckBlock();
 
+        setTargetBufferSize(decodedBlockNode, partitionBufferCapacity);
```

Comment:
I suggest to inline setTargetBufferSize. It is used only once and doesn't actually *set* target size.